src/pages/[locale]/[...path]/index.astro

```diff
@@ -1,26 +1,19 @@
 ---
 import { datocmsCollection, datocmsRequest } from '@lib/datocms';
-import type { PageQuery, PageRecord, ParentPageFragment, SiteLocale } from '@lib/types/datocms';
+import { type PageRouteForPath, getPagePath, getPageSlugFromPath, getParentPages } from '@lib/routing/page';
+import type { PageQuery, SiteLocale } from '@lib/types/datocms';
 import type { PageUrl } from '@lib/types/page-url';
 import Layout from '@layouts/Default.astro';
 import Blocks from '@blocks/Blocks.astro';
 import type { AnyBlock } from '@blocks/Blocks';
-import {
-  getParentPages,
-  getPagePath,
-  getPageSlugFromPath,
-} from '@blocks/InternalLink';
+
 import { formatBreadcrumb } from '@components/Breadcrumbs';
 import PreviewModeSubscription from '@components/PreviewMode/PreviewModeSubscription.astro';
 import ShareButton from '@components/ShareButton/ShareButton.astro';
 import query from './_index.query.graphql';
 
 export async function getStaticPaths() {
-  interface PagesCollectionItem {
-    _allSlugLocales: Array<{ locale: SiteLocale; value: string }>;
-    parentPage: ParentPageFragment;
-  }
-  const pages = await datocmsCollection<PagesCollectionItem>({
+  const pages = await datocmsCollection<PageRouteForPath>({
     collection: 'Pages',
     fragment: `
```

Comment:
@jurgenbelien should we attempt to `print(PageRouteFragment)` (using `graphql/language/printer`) here?

Reply:
not familiar with that, but yes I was indeed wondering if such a thing existed. However we would be overfetching if we would use PageRouteFragment